crypto/fipsmodule/cpucap/cpu_aarch64.c

```diff
@@ -49,4 +49,44 @@ void handle_cpu_env(uint32_t *out, const char *in) {
   }
 }
 
+#if defined(MAKE_DIT_AVAILABLE) && !defined(OPENSSL_WINDOWS)
+// "DIT" is not recognised as a register name by clang-10 (at least)
+// Register's encoded name is from e.g.
+// https://github.com/ashwio/arm64-sysreg-lib/blob/d421e249a026f6f14653cb6f9c4edd8c5d898595/include/sysreg/dit.h#L286
+#define DIT_REGISTER s3_3_c4_c2_5
+
+uint64_t armv8_get_dit(void) {
+  uint64_t val = 0;
+  if (CRYPTO_is_ARMv8_DIT_capable()) {
+    __asm__ volatile("mrs %0, s3_3_c4_c2_5" : "=r" (val));
+  }
+  return val >> 24;
+}
+
+// See https://github.com/torvalds/linux/blob/53eaeb7fbe2702520125ae7d72742362c071a1f2/arch/arm64/include/asm/sysreg.h#L82
+// As per Arm ARM for v8-A, Section "C.5.1.3 op0 == 0b00, architectural hints,
+// barriers and CLREX, and PSTATE access", ARM DDI 0487 J.a, system instructions
+// for accessing PSTATE fields have the following encoding
+// and C5.2.4 DIT, Data Independent Timing:
+//	Op0 = 0, CRn = 4
+//	Op1 (3 for DIT) , Op2 (5 for DIT) encodes the PSTATE field modified and defines the constraints.
+//	CRm = Imm4 (#0 or #1 below)
+//	Rt = 0x1f
+uint64_t armv8_enable_dit(void) {
+  uint64_t original_dit = armv8_get_dit();
+  if (CRYPTO_is_ARMv8_DIT_capable() && original_dit != 1) {
```

Comment:
do we have to check `original_dit != 1`?

Reply:
Removed. Thanks.